### PR TITLE
Fix Handle Parameter and possibility of empty response

### PR DESCRIPTION
**Pull request by nordinvannes**

### CODE REVIEW

File: scripter/connection.go [hunk 1]
```diff
@@ -9,16 +9,20 @@ type ConnectionStruct struct {
 // Handle incoming message string
 // Get all scripts for a given service and pass the string to each script
 func (w *ConnectionStruct) Handle(message string) (string, error) {
-	result := message
+	var result *Result
 	var err error
 
-	result, err = w.MyConn.HandleScripts(w.Service, result)
+	result, err = w.MyConn.HandleScripts(w.Service, message)
```
Comment (Icety) [inline scripter/connection.go]: Why not make it result, err := instead of = and initializing them both first?

File: scripter/lua/connection.go [hunk 2]
```diff
@@ -76,20 +79,21 @@ func (c *luaConn) SetBasicMethods(service string) {
 	}, service)
 
 	c.SetStringFunction("getFileDownload", func() string {
-		url, _ := c.GetParameter(-1, service)
-		path, _ := c.GetParameter(0, service)
+		keys := []string{"url", "path"}
+		params, _ := c.GetParameters(keys, service)
```
Comment (Icety) [inline scripter/lua/connection.go]: Why not put the keys inline instead of declaring a variable?

**Review by Icety — APPROVED** [review]

File: scripter/connection.go (hunk 1 above)
Comment (nordinvannes) [reply scripter/connection.go]: Fixed

File: scripter/lua/connection.go (hunk 2 above)
Comment (nordinvannes) [reply scripter/lua/connection.go]: Fixed

**Pull request merged**